rc/extra/autorestore.kak

```diff
@@ -7,31 +7,42 @@ def autorestore-restore-buffer -docstring "Restore the backup for the current fi
         buffer_basename="${kak_buffile##*/}"
         buffer_dirname=$(dirname "${kak_buffile}")
 
-        ## Find the name of the latest backup created for the buffer that was open
-        ## NB. kak doesn't automatically cd to current buffer_dirname.
-        buffer_backup_glob="${buffer_dirname}/.${buffer_basename}.kak.*"
-        backup_path=$(ls -1t ${buffer_backup_glob} 2>/dev/null | head -n 1)
+        buffer_backup_glob=".${buffer_basename}.kak.*"
 
-        if [ -z "${backup_path}" ]; then
-            exit
+       	if [ -f "${kak_buffile}" ]; then
+       		newer=$(find "${buffer_dirname}" \! -path "${buffer_dirname}" -prune -name "${buffer_backup_glob}" -newer "${kak_buffile}" 2>/dev/null | head -n1)
```

Comment:
I think instead of `\! -path "${buffer_dirname}"` it would be clearer to use `-type f`. Also, I dont think find guarantees the first path printed is going to be the newest. Unfortunately I cant find a way to tell posix find to do that, so I think you'd still need to go through `ls -t -1`.

I'd guess something like `find ... | xargs ls -t -1 | head -n1` will do.

Reply:
`\! -path .. -prune` is to avoid finding the wrong backup file in a subdirectory when we have:
```
./readme.txt
./.readme.txt.kak.z48y
./doc/readme.txt
./doc/.readme.txt.kak.z48y
```